### PR TITLE
Feature: cf create component

### LINKED ISSUE #443: Integrate conditional feature: CF_createComponentFromTemplate
## Description

Integrate conditional feature to create a component from a template.

When it's forbidden: The expanded item of template component in DrawView still open/close, but display no component and display a text that you are not allowed to create component from template.

Role: `CF_createComponentFromTemplate`

add in `src/boot/vue-simple-acl.js`:
```js
setRule('create-template-component', (user) => !process.env.HAS_BACKEND || user.roles.includes('CF_createComponentFromTemplate'));
```

And modify html.

### LINKED ISSUE #444: Integrate conditional feature: CF_createComponent
## Description

Integrate conditional feature to create a component.

When it's forbidden: The expanded item of component in DrawView still open/close, but display no component and display a text that you are not allowed to create component.

Role: `CF_createComponent`

add in `src/boot/vue-simple-acl.js`:
```js
setRule('create-component', (user) => !process.env.HAS_BACKEND || user.roles.includes('CF_createComponent'));
```

And modify html.

### PR DESCRIPTION
**Pull request by Zorin95670**
Resolved #443
Resolved #444

### CODE REVIEW

File: src/boot/vue-simple-acl.js [hunk 1]
```diff
@@ -8,12 +8,12 @@ const userRoles = computed(() => ({ roles: userStore.roles }));
 const rules = () => defineAclRules((setRule) => {
   setRule('create-diagram', (user) => !process.env.HAS_BACKEND || user.roles.includes('CF_createDiagram'));
   setRule('create-diagram-from-template', (user) => !process.env.HAS_BACKEND || user.roles.includes('CF_createDiagramFromTemplate'));
+  setRule('create-template-diagram', (user) => !process.env.HAS_BACKEND || user.roles.includes('CF_createDiagramFromTemplate'));
```
Comment (k-y0u) [inline src/boot/vue-simple-acl.js:11]: wrong c/p

Comment (Zorin95670) [reply src/boot/vue-simple-acl.js]: done sorry for bad rebase!

File: src/boot/vue-simple-acl.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ const userRoles = computed(() => ({ roles: userStore.roles }));
 const rules = () => defineAclRules((setRule) => {
   setRule('create-diagram', (user) => !process.env.HAS_BACKEND || user.roles.includes('CF_createDiagram'));
   setRule('create-diagram-from-template', (user) => !process.env.HAS_BACKEND || user.roles.includes('CF_createDiagramFromTemplate'));
+  setRule('create-component', (user) => !process.env.HAS_BACKEND || user.roles.includes('CF_createComponent'));create-component rule in acl)
```
Comment (k-y0u) [inline src/boot/vue-simple-acl.js:11]: problem : `user.roles.includes('CF_createComponent'));create-component rule in acl)`

Comment (aboubakar14) [inline src/boot/vue-simple-acl.js:11]: wrong c/p again, at the end

File: src/components/list/LibraryList.vue [hunk 3]
```diff
@@ -17,34 +17,45 @@
...
-      style="overflow-y: auto !important;"
-      class="q-pa-none q-mb-sm"
-    >
-      <component-definition-grid
-        :definitions="selectedItemDefinitions"
-        :plugin-name="plugin.data.name"
-        class="q-mt-xs q-pr-sm"
-      />
-    </q-item>
+    <template v-if="$acl.role('create-component')">
```
Comment (k-y0u) [inline src/components/list/LibraryList.vue:33]: Add a `<template>` for `v-if`, why not. But why not add one for `v-else-if` ?

File: src/components/list/LibraryList.vue [hunk 4]
```diff
@@ -17,34 +17,45 @@
...
+      <q-item
+        v-if="selectedItemId && !isEmptyList"
+        style="overflow-y: auto !important;"
```
Comment (k-y0u) [inline src/components/list/LibraryList.vue:36]: in `<style>`

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/boot/vue-simple-acl.js (hunk 2 above)
Comment (Zorin95670) [reply src/boot/vue-simple-acl.js]: done

File: src/components/list/LibraryList.vue (hunk 3 above)
Comment (Zorin95670) [reply src/components/list/LibraryList.vue]: rework on all code

File: src/components/list/LibraryList.vue (hunk 4 above)
Comment (Zorin95670) [reply src/components/list/LibraryList.vue:43]: rework on all code, but don't understand your comment.

**Review by aboubakar14 — APPROVED** [review]

**Review by k-y0u — APPROVED** [review]

**Pull request merged**